ethportal-api/src/types/portal_wire.rs

```diff
@@ -174,22 +176,41 @@ pub enum ProtocolId {
     Utp,
 }
 
-/// Encode hex string to protocol id
-impl FromStr for ProtocolId {
-    type Err = ProtocolIdError;
-
-    fn from_str(input: &str) -> Result<ProtocolId, Self::Err> {
-        match input {
-            "0x500A" => Ok(ProtocolId::State),
-            "0x500B" => Ok(ProtocolId::History),
-            "0x500C" => Ok(ProtocolId::TransactionGossip),
-            "0x500D" => Ok(ProtocolId::CanonicalIndices),
-            "0x501A" => Ok(ProtocolId::Beacon),
-            "0x757470" => Ok(ProtocolId::Utp),
-            _ => Err(ProtocolIdError::Invalid),
-        }
-    }
-}
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct NetworkSpec {
+    pub network_name: String,
+    pub portal_networks: BiHashMap<ProtocolId, String>,
+}
+
+pub static MAINNET: Lazy<Arc<NetworkSpec>> = Lazy::new(|| {
+    let mut portal_networks = BiHashMap::new();
+    portal_networks.insert(ProtocolId::State, "0x500A".to_string());
+    portal_networks.insert(ProtocolId::History, "0x500B".to_string());
+    portal_networks.insert(ProtocolId::TransactionGossip, "0x500C".to_string());
+    portal_networks.insert(ProtocolId::CanonicalIndices, "0x500D".to_string());
+    portal_networks.insert(ProtocolId::Beacon, "0x501A".to_string());
+    portal_networks.insert(ProtocolId::Utp, "0x757470".to_string());
+    NetworkSpec {
+        portal_networks,
+        network_name: "mainnet".to_string(),
+    }
+    .into()
+});
+
+pub static TESTNET: Lazy<Arc<NetworkSpec>> = Lazy::new(|| {
+    let mut portal_networks = BiHashMap::new();
+    portal_networks.insert(ProtocolId::State, "0x501A".to_string());
+    portal_networks.insert(ProtocolId::History, "0x501B".to_string());
+    portal_networks.insert(ProtocolId::TransactionGossip, "0x501C".to_string());
```

Comment:
I don't see any reason to include `TransactionGossip` & `CanonicalIndices`?

Reply:
I mean it is included in our mainnet code